src/hints/mod.rs

```diff
@@ -336,3 +339,24 @@ pub fn breakpoint(
     println!("\n-----------END BREAKPOINT-----------\n");
     Ok(())
 }
+
+const IS_ON_CURVE: &str = "ids.is_on_curve = (y * y) % SECP_P == y_square_int";
+pub fn is_on_curve(
+    vm: &mut VirtualMachine,
+    exec_scopes: &mut ExecutionScopes,
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+    _constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError> {
+    let y: BigInt = exec_scopes.get("y")?;
+    let y_square_int: BigInt = exec_scopes.get("y_square_int")?;
+    // TODO: assume SECP_P is in scope, or should we use this constant?
+    // let SECP_P = cairo_vm::hint_processor::builtin_hint_processor::secp::secp_utils::SECP_P;
+    let SECP_P: BigInt = exec_scopes.get("SECP_P")?;
```

Comment:
It's not clear from the hint whether this should have been injected into `exec_scopes` from some other code/hint or if it should just exist implicitly.

In any case, the constant mentioned (`cairo_vm::hint_processor::builtin_hint_processor::secp::secp_utils::SECP_P`) is private.

Reply:
From other hints defined in `cairo-vm` ([see here](https://github.com/lambdaclass/cairo-vm/blob/dea2723d99da9bcf4c79e77044ccef4891c1cc9a/vm/src/hint_processor/builtin_hint_processor/secp/field_utils.rs#L37)) it seems that SECP_P should be inserted at some point in the execution scope but is also a constant in the VM.